sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import java.util.Locale
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, CodeGenerator, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.codegen.Block.BlockHelper
+import org.apache.spark.sql.catalyst.util.NumberUtils.NumberFormatBuilder
+import org.apache.spark.sql.types.{DataType, DecimalType, StringType}
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * A function that converts string to numeric.
+ */
+@ExpressionDescription(
+  usage = """
+     _FUNC_(strExpr, formatExpr) - Convert `strExpr` to a number based on the `formatExpr`.
+       The format can consist of the following characters:
+         '9':  digit position (can be dropped if insignificant)
+         '0':  digit position (will not be dropped, even if insignificant)
+         '.':  decimal point (only allowed once)
+         ',':  group (thousands) separator
+         'S':  sign anchored to number (uses locale)
+         'D':  decimal point (uses locale)
```

Comment:
is it true? We simply relace `D` with `.`, and I don't see we handle locale there.

Reply:
Please see https://github.com/apache/spark/blob/5cfb5dfde5e566a4380def46324034604012af32/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberUtils.scala#L172

Reply:
I only see that we replace `D` with `.` during normalization.

Reply:
This comment references PostgreSQL and Oracle. But other database does not mention locale.
I guess `NumberFormat.getNumberInstance(Locale.ROOT)` is enough.

Reply:
the Spark doc should match Spark implementation

Reply:
are D, S, G commonly supported in other databases? If not, we can remove them as we don't support locale.